src/core/api/index.js

```diff
@@ -62,27 +63,35 @@ export function callApi({
 
   return fetch(apiURL, options)
     .then((response) => {
-      return response.clone().json()
-        .then(
-          (jsonResponse) => ({ response, jsonResponse }),
-          (error) => {
+      function makeTextResponseJson(textResponse) {
+        return { jsonResponse: { text: textResponse }, response };
+      }
+
+      if (response.headers.get('Content-Type') === 'application/json') {
+        return response.json()
+          .then((jsonResponse) => ({ response, jsonResponse }))
+          .catch((error) => {
             log.warn('Could not parse response as JSON:', error);
-            return response.clone().text().then((textResponse) =>
-              ({ response, jsonResponse: { text: textResponse } })
-            );
-          }
-        );
+
+            return response.text()
```

Comment:
This is going to fail with a TypeError because the stream is drained at this point. Until we process the stream in chunks, we don't have a way to fix it. I say, for now, just let `response.json()` throw an error. Maybe it would be nice to catch the error, log the response headers, and then re-throw it.